packages/buidler-etherscan/src/solc/bytecode.ts

```diff
@@ -0,0 +1,260 @@
+import { RunTaskFunction } from "@nomiclabs/buidler/types";
+
+import { metadataLengthSize, readSolcMetadataLength } from "./metadata";
+import { InferralType } from "./version";
+
+export async function lookupMatchingBytecode(
+  contractFiles: CompilerOutput["contracts"],
+  deployedBytecode: string,
+  inferralType: InferralType
+) {
+  for (const [contractFilename, contracts] of Object.entries(contractFiles)) {
+    for (const [contractName, contract] of Object.entries(contracts)) {
+      // Normalize deployed bytecode according to this contract.
+      const { deployedBytecode: runtimeBytecodeSymbols } = contract.evm;
+
+      const comparison = await compareBytecode(
+        deployedBytecode,
+        runtimeBytecodeSymbols,
+        inferralType
+      );
+
+      if (comparison.match) {
+        const {
+          contractInformation: {
+            immutableValues,
+            libraryLinks,
+            normalizedBytecode,
+          },
+        } = comparison;
+        // The bytecode matches
+        return {
+          immutableValues,
+          libraryLinks,
+          normalizedBytecode,
+          contractFilename,
+          contractName,
+          contract,
+        };
+      }
+    }
+  }
+
+  return null;
+}
+
+type BytecodeComparison =
```

Comment:
I don't understand why this is here, it isn't used anywhere and just adds complexity

Reply:
ok, I see. this is used in `lookupMatchingBytecode`. It's really weird that the type is defined below the place where it's used.

Anyway, a more common way of expressing this would be `BytecodeExtractedData | undefined`